pvlib/tracking.py

```diff
@@ -569,3 +582,163 @@ def singleaxis(apparent_zenith, apparent_azimuth,
+
+        tan(tr_ze) = -tan(sys_ze)*cos(sys_az-tr_az)
+    """
+    sys_az_rel_to_tr_az = system_azimuth - tracker_azimuth
+    tan_tr_ze = -np.cos(sys_az_rel_to_tr_az) * np.tan(system_zenith)
+    return -np.arctan(tan_tr_ze)
+
+
+def calc_system_tracker_side_slope(
+        tracker_azimuth, tracker_zenith, system_azimuth, system_zenith):
+    """
+    Calculate the slope perpendicular to the tracker axis relative to the
+    system plane containing the axes as well as the rotation of the tracker
+    axes relative to the system plane. Note in order for the backtracking
+    algorithm to work correctly on a sloped system plane, the side slope must
+    be applied to the tracker rotation.
+
+    Parameters
+    ----------
+    system_azimuth : float
+        direction of normal to slope on horizontal [radians]
+    system_zenith : float
+        tilt of normal to slope relative to vertical [radians]
+    tracker_azimuth : float
+        direction of tracker axes on horizontal [radians]
+    tracker_zenith : float
+        tilt of tracker [radians]
+
+    Returns
+    -------
+    tracker side slope and rotation relative to system plane [radians]
+    """
+    # find the relative rotation of the trackers in the system plane
+    # 1. tracker axis vector
+    cos_tr_ze = np.cos(-tracker_zenith)
+    sin_tr_az = np.sin(tracker_azimuth)
+    cos_tr_az = np.cos(tracker_azimuth)
+    tr_ax = np.array([
+        [cos_tr_ze*sin_tr_az],
+        [cos_tr_ze*cos_tr_az],
+        [np.sin(-tracker_zenith)]])
+    # 2. rotate tracker axis vector from global to system reference frame
+    sys_z_rot = _get_rotation_matrix(system_azimuth, axis=2)
+    # first around the z-axis
+    tr_ax_sys_z_rot = np.dot(sys_z_rot, tr_ax)
+    # then around x-axis so that xy-plane is the plane with slope and trackers
+    sys_x_rot = _get_rotation_matrix(system_zenith)
+    tr_ax_sys = np.dot(sys_x_rot, tr_ax_sys_z_rot)
+    # now that tracker axis is in coordinate system of slope, the relative
+    # rotation is the angle from the y axis
+    tr_rel_rot = np.arctan2(tr_ax_sys[0, 0], tr_ax_sys[1, 0])
+    # find side slope
+    # 1. tracker normal vector
+    sin_tr_ze = np.sin(tracker_zenith)
+    tr_norm = np.array([
+        [sin_tr_ze*sin_tr_az],
+        [sin_tr_ze*cos_tr_az],
+        [cos_tr_ze]])  # note: cos(-x) = cos(x)
+    # 2. rotate tracker normal vector from global to system reference frame
+    tr_norm_sys_z_rot = np.dot(sys_z_rot, tr_norm)
+    tr_norm_sys = np.dot(sys_x_rot, tr_norm_sys_z_rot)
+    # 3. side slope is angle between tracker normal and system plane normal
+    # np.arccos(tr_norm_sys[2])
+    # 4. but we need to know which way the slope is facing, so rotate to
+    # tracker use arctan2
+    sys_tr_z_rot = _get_rotation_matrix(tr_rel_rot, axis=2)
+    tr_norm_sys_tr = np.dot(sys_tr_z_rot, tr_norm_sys)
+    side_slope = np.arctan2(tr_norm_sys_tr[0, 0], tr_norm_sys_tr[2, 0])
+    return side_slope, tr_rel_rot
```

Comment:
I'm not immediately seeing the utility of `tr_rel_rot` -- how would it be used?

Reply:
It's the Z-rotation you mentioned between the tracker axes and the azimuth of the system/slope plane normal. [It's an intermediate output](../blob/5dc2c5479e064540ba435ae149015eeb9925dd41/pvlib/tracking.py#L726), so I figured why not return it, but it's not really useful for anything. We just need it [to determine which direction the side-slope is facing](../blob/5dc2c5479e064540ba435ae149015eeb9925dd41/pvlib/tracking.py#L737-L741). My mild preference is to keep it, but I'm okay with removing it too.